src/google/protobuf/parse_context.h

```diff
@@ -428,7 +428,7 @@ class PROTOBUF_EXPORT ParseContext : public EpsCopyInputStream {
 template <uint32 tag>
 bool ExpectTag(const char* ptr) {
   if (tag < 128) {
-    return *ptr == tag;
+    return *ptr == tag % 128;
```

Comment:
This seems a bit over-subtle... how about just a `static_cast<char>(tag)`?

Reply:
Sure, whatever.